api_key_stamper/src/lib.rs

```diff
@@ -1,122 +1,383 @@
-        signature: hex::encode(sig.to_der()),
-        scheme: "SIGNATURE_SCHEME_TK_API_P256".to_string(),
-    };
+    /// Load an API key from hex-encoded strings
+    /// Optionally takes in public a public key to check against, after derivation.
+    /// If the derived public key doesn't match the passed in value, `PublicKeyMismatch` is returned.
+    pub fn from_strings<S: AsRef<str>>(
+        private_key: S,
+        public_key: Option<S>,
+    ) -> Result<Self, StamperError> {
+        let private_key_bytes = hex::decode(private_key.as_ref())
+            .map_err(|e| StamperError::HexDecode(e.to_string()))?;
 
-    let json_stamp = serde_json::to_string(&stamp).unwrap();
+        let public_key_bytes = if public_key.is_some() {
+            Some(
+                hex::decode(public_key.unwrap().as_ref())
+                    .map_err(|e| StamperError::HexDecode(e.to_string()))?,
+            )
+        } else {
+            None
+        };
+
+        Self::from_bytes(private_key_bytes, public_key_bytes)
+    }
+
+    /// Helper to create an API from pre-existing files. This is useful if you used
+    /// the Turnkey CLI (<https://github.com/tkhq/tkcli>) to generate keys.
+    pub fn from_files<P: AsRef<Path>, Q: AsRef<Path>>(
+        private_key_path: P,
+        public_key_path: Q,
+    ) -> Result<Self, StamperError> {
+        let private_key = fs::read_to_string(private_key_path.as_ref()).map_err(|e| {
+            StamperError::Io(private_key_path.as_ref().to_path_buf(), e.to_string())
+        })?;
+        let public_key = fs::read_to_string(public_key_path.as_ref())
+            .map_err(|e| StamperError::Io(public_key_path.as_ref().to_path_buf(), e.to_string()))?;
+
+        let private_key_hex_bytes =
+            hex::decode(private_key).map_err(|e| StamperError::HexDecode(e.to_string()))?;
+        let public_key_hex_bytes =
+            hex::decode(public_key).map_err(|e| StamperError::HexDecode(e.to_string()))?;
+
+        Self::from_bytes(private_key_hex_bytes, Some(public_key_hex_bytes))
+    }
+
+    /// Function to produce a base64-encoded stamp.
+    /// See <https://docs.turnkey.com/developer-reference/api-overview/stamps#api-keys> for more information.
+    pub fn stamp<S: AsRef<str>>(&self, request_body: S) -> Result<String, StamperError> {
+        let sig: Signature = self.signing_key.sign(request_body.as_ref().as_bytes());
+
+        let stamp = TurnkeyApiStamp {
+            public_key: hex::encode(self.compressed_public_key()),
+            signature: hex::encode(sig.to_der()),
+            scheme: "SIGNATURE_SCHEME_TK_API_P256".to_string(),
+        };
+
+        let json_stamp = serde_json::to_string(&stamp).unwrap();
+
+        Ok(BASE64_URL_SAFE_NO_PAD.encode(json_stamp.as_bytes()))
+    }
 
-    Ok(BASE64_URL_SAFE_NO_PAD.encode(json_stamp.as_bytes()))
+    /// Returns the compressed public key bytes
+    pub fn compressed_public_key(&self) -> Vec<u8> {
```

Comment:
We might need the an uncompressed representation as well at some point?

Reply:
Yep! If we need it we can add that later. But right now I'd rather keep the API surface small. Bonus: minimal confusion because there's only one way to get a public key :)